src/partial-assembly.md

```diff
@@ -0,0 +1,174 @@
+of freedom on each element*, transitions to independent *quadrature points* in
+reference space, performs the integration, and then goes back in reverse order
+to global (test) degrees of freedom on the whole mesh.
+
+This is illustrated below for the case of a symmetric linear operator. We use
+the notions **T-vector**, **L-vector**, **E-vector** and **Q-vector** to
+represent the sets corresponding to the (true) degrees of freedom on the global
+mesh, the split local degrees of freedom on the subdomains, the split degrees
+of freedom on the mesh elements, and the values at quadrature points,
+respectively.
+
+We refer to the operators that connect the different types of vectors as:
+
+- Subdomain restriction **P**
+- Element restriction **G**
+- Basis (Dofs-to-Qpts) evaluator **B**
+- Operator at quadrature points **D**
+
+More generally, when the test and trial space differ, they get their own
+versions of **P**, **G** and **B**.
+
+![Operator Decomposition](img/libceed.png "Operator Decomposition")
+
+Note that in the case of adaptive mesh refinement (AMR), the prolongation
+operator **P** involves not only extracting sub-vectors, but evaluating values
+at constrained degrees of freedom through the AMR interpolation. There can also
+be several levels of subdomains (**P1**, **P2**, etc.), and it may be
+convenient to split **D** as the product of several operators (**D1**, **D2**,
+etc.).
+
+## Partial Assembly in MFEM
+
+Since the global operator **A** is just a series of variational restrictions
+with **B**, **G** and **P**, starting from its point-wise kernel **D**, a
+"matrix-vector product" with **A** can be performed by evaluating and storing
+some of the innermost variational restriction matrices, and applying the rest
+of the operators "on-the-fly". For example, one can compute and store a global
+matrix on T-vector level. Alternatively, one can compute and store only the
+subdomain (L-vector) or element (E-vector) matrices and perform the action of
+**A** using matvecs with **P** or **P** and **G**. While these options are
+natural for low-order discretizations, they are not a good fit for high-order
+methods due to the amount of FLOPs needed for their evaluation, as well as the
+memory transfer needed for a matvec.
+
+MFEM's **partial assembly** functionality computes and stores only **D** (or
+portions of it) and evaluates the actions of **P**, **G** and **B** on-the-fly.
+Critically for performance, MFEM takes advantage of the tensor-product
+structure of the degrees of freedom and quadrature points on quadrilateral and
+hexahedral elements to perform the action of **B** without storing it as a
+matrix. Note that the action of **B** is performed elementwise (it corresponds
+to a block-diagonal matrix), and the blocks depend only on the element order
+and reference geometry. Currently, only fixed order and geometry is supported,
+meaning that all the blocks of **B** are identical.
```

Comment:
```suggestion
and reference geometry. In the common case where the mesh has fixed polynomial order
and only one element type, then all the blocks of **B** are identical.
```

Reply:
Should we not also mention that this common case is the only one that is currently supported (natively) in MFEM?

Reply:
I was hoping to leave the documentation generic, so that it still applies if this capability is added, but that's probably overly lazy and you're right that we should mention this.